jsonnet/lib/observatorium-api.libsonnet

```diff
@@ -1,339 +1,392 @@
-    deployment.mixin.spec.selector.withMatchLabels(api.config.podLabelSelector) +
-    deployment.mixin.spec.template.spec.withServiceAccount(api.serviceAccount.metadata.name) +
-    deployment.mixin.spec.strategy.rollingUpdate.withMaxSurge(0) +
-    deployment.mixin.spec.strategy.rollingUpdate.withMaxUnavailable(1) +
-    deployment.mixin.spec.template.spec.withVolumes(
-      (if api.config.rbac != {} then [
-         {
-           configMap: {
-             name: api.config.name,
-           },
-           name: 'rbac',
-         },
-       ] else []) +
-      (if api.config.tenants != {} then [
-         {
-           secret: {
-             secretName: api.config.name,
-           },
-           name: 'tenants',
-         },
-       ] else []) +
-      (if std.objectHas(api.config.tenants, 'tenants') then [
-        if std.objectHas(tenant.mTLS, 'secretName') then {
-          secret: {
-            secretName: tenant.mTLS.secretName,
-          },
-          name: tenant.name + '-mtls-secret',
-        } else if std.objectHas(tenant.mTLS, 'configMapName') then {
-          confiMap: {
-            name: tenant.mTLS.configMapName,
-          },
-          name: tenant.name + '-mtls-configmap',
+  deployment: {
+    apiVersion: 'apps/v1',
+    kind: 'Deployment',
+    metadata: {
+      name: api.config.name,
+      namespace: api.config.namespace,
+      labels: api.config.commonLabels,
+    },
+    spec: {
+      replicas: api.config.replicas,
+      selector: { matchLabels: api.config.podLabelSelector },
+      strategy: {
+        rollingUpdate: {
+          maxSurge: 0,
+          maxUnavailable: 1,
         },
-        for tenant in api.config.tenants.tenants
-        if std.objectHas(tenant, 'mTLS')
-       ] else []) +
-      (if api.config.tls != {} then [
-         {
-           secret: {
-             secretName: api.config.tls.secretName,
-           },
-           name: 'tls-secret',
-         },
-       ] + (
-         if std.objectHas(api.config.tls, 'configMapName') then [
-           {
-             configMap: {
-               name: api.config.tls.configMapName,
-             },
-             name: 'tls-configmap',
-           },
-         ] else []
-       ) else [])
-    ),
-
-  configmap:
-    if api.config.rbac != {} then {
-      apiVersion: 'v1',
-      data: {
-        'rbac.yaml': std.manifestYamlDoc(api.config.rbac),
       },
-      kind: 'ConfigMap',
-      metadata: {
-        labels: api.config.commonLabels,
-        name: api.config.name,
-        namespace: api.config.namespace,
-      },
-    } else null,
-
-  secret:
-    if api.config.tenants != {} then {
-      local tenants = {
-        tenants: [
-          {
-            id: tenant.id,
-            name: tenant.name,
-            mTLS: {
-              caPath: '/var/run/mtls/' + tenant.name + '/' + tenant.mTLS.caKey,
+      template: {
+        metadata: { labels: api.config.commonLabels },
```

Comment:
shouldn't this be `api.config.podLabelSelector`, I found service has selector of `api.config.podLabelSelector`.

Reply:
@morvencao
I use this one: https://github.com/observatorium/api/blob/main/jsonnet/lib/observatorium-api.libsonnet
only change I made on it the the tenant-header part